orchagent/response_publisher.h

```diff
@@ -57,8 +61,29 @@ class ResponsePublisher : public ResponsePublisherInterface
     void setBuffered(bool buffered);
 
   private:
+    struct entry
+    {
+        std::string table;
+        std::string key;
+        std::vector<swss::FieldValueTuple> values;
+        std::string op;
+        bool replace;
+        bool flush;
+        bool shutdown;
+    };
+
+    void dbUpdateThread();
+    void writeToDBInternal(const std::string &table, const std::string &key,
+                           const std::vector<swss::FieldValueTuple> &values, const std::string &op, bool replace);
+
     std::unique_ptr<swss::DBConnector> m_db;
+    std::unique_ptr<swss::RedisPipeline> m_ntf_pipe;
     std::unique_ptr<swss::RedisPipeline> m_pipe;
 
     bool m_buffered{false};
+    // Thread to write to DB.
+    std::unique_ptr<std::thread> m_update_thread;
+    std::queue<entry> m_queue;
+    mutable std::mutex m_lock;
```

Comment:
>mutable [](http://example.com/codeflow?start=4&length=7)

why mutable?

Reply:
That is a common practice for mutex object.

The key word "mutable" means that the variable can be changed in a const method.
If we have a method to read a variable that is protected by a mutex, it is nature to declare the method as const since it only does read operation. But the mutex object needs to be mutable for the read method to get the lock.

In this case, we might not really need it to be mutable. But we should follow the common practice.

Reply:
we might not really need it to be mutable -> let's just remove it.

Thanks for the explanation!